client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoResultSet.java

```diff
@@ -129,7 +129,7 @@ private static List<Column> getColumns(StatementClient client, Consumer<QuerySta
             QueryStatusInfo results = client.currentStatusInfo();
             progressCallback.accept(QueryStats.create(results.getId(), results.getStats()));
             List<Column> columns = results.getColumns();
-            if (columns != null) {
+            if (columns != null && !results.getStats().getState().equals("FAILED")) {
```

Comment:
Do you know how does this get empty columns? @wendigo 
I think it's a bug that should fix in the place that return empty columns